tests/shared/grub2.sh

```diff
@@ -1,6 +1,16 @@
-GRUB_CFG_ROOT=/boot/grub2
+test -n "$GRUB_CFG_ROOT" || GRUB_CFG_ROOT=/boot/grub2
+
+function set_grub_uefi_root {
+	if grep NAME /etc/os-release | grep -iq fedora; then
+		GRUB_CFG_ROOT=/boot/efi/EFI/fedora
+	else
+		GRUB_CFG_ROOT=/boot/efi/EFI/redhat
+	fi
+}
 
 function make_grub_password {
+	mkdir -p "$GRUB_CFG_ROOT"
+	set_superusers "root"
```

Comment:
The GRUB password rule checks that admin username is a valid username, so this is needed. It also creates the `grub.cfg` file - without it, the rule would pass due to not being applicable.